src/Sylius/Bundle/ApiBundle/Controller/Payment/GetPaymentConfiguration.php

```diff
@@ -31,13 +31,19 @@ public function __construct(
 
     public function __invoke(Request $request): JsonResponse
     {
-        /** @var PaymentInterface|null $payment */
-        $payment = $this->paymentRepository->findOneByOrderToken(
-            $request->attributes->get('paymentId'),
-            $request->attributes->get('tokenValue'),
-        );
-
-        Assert::notNull($payment);
+        try {
+            /** @var PaymentInterface|null $payment */
+            $payment = $this->paymentRepository->findOneByOrderToken(
+                $request->attributes->get('paymentId'),
+                $request->attributes->get('tokenValue'),
+            );
+        } catch (\Exception) {
```

Comment:
Again, catching a generic exception is not good in this place.

Reply:
I believe that many of these `id`-ish resources could be simplified with APIP3 sub-resources, but for now we have to settle for what we have